lib/storage/datauri.js

```diff
@@ -27,7 +28,33 @@ async function download(source, file) {
         throw new GenericError(err.message, `${actionName()}_download`);
     }
 }
+/**
+ * Generate Presign Url taking local file path as input
+ * Creates a unqiue name for the file, upload the file to 
+ * storage from the path provided and generates a PreSigned Url
+ * @param {String} Path to local file system, where file is downloaded
+ * @returns {String} PreSigned Url of upload the source file
+ */
+async function getPreSignedUrl(source) {
```

Comment:
After the download, or before arriving here, before generating thr URL, please verify that:
- the file is indeed available;
- the file size is not 0 (empty file).
If any of those is true, we should throw an error. I would choose SourceUnsupportedError, but it's up to debate.

Reply:
Added the scenarios and threw SourceUnsupportedError, however caught the error in storage.js and logged it as this should not prevent further execution, this is nice to have for downstream services but worker as is may not need it

Reply:
Threw SourceUnSupportedError and did not catch in the caller